### PR TITLE
fix overscan issue and enable linkifier on partial shown matches

### LINKED ISSUE #1691: linkifier - match urls at the viewport borders
The new linkifier currently does not match partially shown urls at the viewport borders.
Should be fixable by allowing some "overscan" at the top and the bottom of the viewport.

### PR DESCRIPTION
**Pull request by jerch**
Fixes #1691.

@Tyriar I actually had to move the overscan ability to `BufferStringIterator` to deal with a possible worst case scenario where the buffer contains very very long unwrapped lines.
This might also be a problem at other places that try to unwrap line data.

~The linkifier now has a class attribute `LIMIT_OVERSCAN = 5` that allows to expand the unwrapping to a maximum of -5 for the top and +5 for the bottom viewport border. I think we want this value to be customizable or changeable at runtime, since it kinda limits any match at the borders to that size (an URI taking 6 lines will not match at the borders), which will lead to problems in small terminals. Not sure where to put this, maybe in the buffer class itself as an default overscan value, that changes with different buffer widths?~ Changed to class attribute `LIMIT_CHAR_OVERSCAN = 2000`, that does the overscan based on the actual terminal width.

### CODE REVIEW

**Comment by Tyriar** [comment]
> I think we want this value to be customizable or changeable at runtime, since it kinda limits any match at the borders to that size (an URI taking 6 lines will not match at the borders), which will lead to problems in small terminals.

Why not just ballpark some number and accept the few failures that may occur? For example, scan 5 rows up or down, or maybe better scan x characters up or down (`Math.ceil(charCount / cols)` rows).

**Comment by jerch** [comment]
Right x chars before and after will do and automagically deal with different buffer widths. :+1:

**Comment by jerch** [comment]
According [this informed SO post](https://stackoverflow.com/a/417184) URLs longer than 2k are not widely supported, therefore I changed the overscan limit to 2000 characters.

**Comment by jerch** [comment]
@Tyriar: Ready for next review. Lemme know if something is still missing.

File: src/Buffer.test.ts [hunk 1]
```diff
@@ -514,4 +514,35 @@ describe('Buffer', () => {
...
+        'aaaaaaaaaa',
+        'aaaaaaaaaa',
+        'aaaaaaaaaa',
+        'aaaaaaaaa\n',
+        'aaaaaaaaaa',
+        'aaaaaaaaaa'
+      ];
+      terminal.writeSync(data.join(''));
+      // brute force test with insane values
+      expect(() => {
```
[the remark this answers is not in the document]
Comment (jerch) [reply src/Buffer.test.ts]: Sweet, yeah took me some minutes to figure this one out.

**Comment by jerch** [comment]
@Tyriar Thx alot :heart:

**Pull request merged**